app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,27 @@
+class OrdersController < ApplicationController
+
+  include CurrentOrderFinder
+  include LineItemsHandler
+
+  before_action :set_current_order
+  before_action :set_deal, only: %i[update]
+  before_action :set_line_item, only: %i[update]
+  skip_before_action :authorize, only: %i[cart update]
+
+  def cart
+    @line_items = @order.line_items.includes(:deal)
+  end
+
+  def update
+    if params[:task] == 'decrement' && @line_item.quantity > 1
```

Comment:
it will be like order.udpate_cart(permitted_params) and rest of the logic will be in model